### DISCUSSION TITLE
automatically capitalize search terms that are used via a shortcut, e.g. to Wikipedia

### DISCUSSION THREAD

**Discussion by ReaderGuy42** [Ulauncher/Ulauncher#1269] [Q&A]
I'm using a Wikipedia shortcut:
![image](https://github.com/Ulauncher/Ulauncher/assets/71895400/7c0097e2-f77a-41ba-addd-0e130b5fad48)

and while it does what it's supposed to, I've encountered a reoccurring problem, where a search term of multiple words isn't found because only the first word is automatically capitalized and the second (or third) words aren't.

Apologies for the gruesome subject matter, this is the example I found and I can't remember the other examples just now.

I launch Ulauncher and then enter `w tommy silverstein` and this launches the URL: `https://en.wikipedia.org/wiki/Tommy_silverstein` which doesn't exist. If I manually correct it to `https://en.wikipedia.org/wiki/Tommy_Silverstein` it automatically redirects to the actual article `https://en.wikipedia.org/wiki/Thomas_Silverstein`.

I'm not sure if the problem I'm having is that Wikipedia only corrects certain aliases, e.g. I can search `w morgan freeman` and that gets corrected to `https://en.wikipedia.org/wiki/Morgan_Freeman` as desired.

But is there a way to have Ulauncher automatically capitalize all words of a search when redirecting to a website?

Thanks!

**Comment by friday** [comment]
This is not something we will support, but you can make your own shell script shortcut or extension to add more advanced functionality like this.
